app/forms/supplier_user_form.rb

```diff
@@ -7,7 +7,10 @@ class SupplierUserForm
   attr_accessor :full_name, :email, :supplier
   validates :supplier, presence: { message: "Select one" }, on: :supplier
   validates :full_name, presence: { message: "Enter a name" }, on: :details
-  validates :email, presence: { message: "Enter email" }, on: :details
+  validates :email,
+            presence: { message: "Enter email" },
+            format: { with: Devise.email_regexp, message: "Enter an email address in the correct format, like name@example.com" },
```

Comment:
I think it will display two error messages for blank email. We need `allow_blank: true` here

Reply:
Doesn't seem to be the case 
![image](https://user-images.githubusercontent.com/40488007/115568012-c69dad00-a2b3-11eb-9b34-a3881522a02d.png)

Reply:
Interesting. So it seems controller is casting `""` into `nil` which is why fromat_validator ignores the check. It'd still happen in the console though:

![image](https://user-images.githubusercontent.com/2413414/115571229-b804c500-a2b6-11eb-941a-7b0feefd8be3.png)

Happy to leave it, it is really more of the nit-pick than an issue :P

Reply:
Huh, ok. Just merged, will update in future if I remember